### ISSUE TITLE
Ionic 2 event with pop page double trouble :) (Ionic 2 RC1)

### ISSUE THREAD

**Issue by azakusilo** [driftyco/ionic#8901]
Hello I got 3 pages.
1 - root, 2 - category page 3 - category filter page
At the third page I got a child component inside where I use event publish. 
```
applyFilters()
  {
    //Вызовем событие применения фильтров
    this.events.publish('category_filters: pop');
    this.events.publish('category_filters: apply', this.category.getFiltersForQuery());
  }
```
I use with function on button click.
In parent page(third) I catch it and try to pop current page after catch.
```
ngOnInit()
  {
    this.events.subscribe('category_filters: pop', (eventsData) => {
      this.navCtrl.pop();
    });
  }
```
At first time I press button for pop page. It's pop third page like it must. At second time press it pop my third page and second page. At third time press it pop all 3 pages.
Why? May be bug?

**Comment by tomgallagher** [comment]
Hi

**Comment by azakusilo** [comment]
Hello

**Comment by azakusilo** [comment]
> Try a console log statement in the listener to see if it's that. It will fire twice if we have the same problem. 

Yes it is the same. My event fires 1 time, then 2 times, then 3 time and etc.

**Comment by jgw96** [comment]
Hello all! Thanks for opening an issue with us! Would you all mind posting a repo we can use to reproduce this issue?

**Comment by azakusilo** [comment]
> Hello all! Thanks for opening an issue with us! Would you all mind posting a repo we can use to reproduce this issue? 

I'm glad. But I don't how to do this. Do you have any example?

**Comment by tomgallagher** [comment]
I'd certainly like to see the code for all three pages to compare to mine. The question is why, when you place the events listener in the constructor, is it being instantiated multiple times? When I create a tab root page, navigate to a child via nav.push, then pop that page and come back, I would expect that page to still be there at the bottom of the stack. At the moment, I think that page is being re-created each time nav.pop is being called and we head to the bottom of the stack.

It could be a bug in the caching of root pages.

There's a discussion on the forum on a related issue here:

https://forum.ionicframework.com/t/events-and-page-lifecycles/59204

**Comment by nalkon** [comment]
Are there any news with this issue?

**Comment by azakusilo** [comment]
No. But I rewrite code, to do not use event on "pop" navChain action.

**Comment by nalkon** [comment]
Well, I ended up by re-writing code without firing event at all.

**Comment by nalkon** [comment]
Installed RC3 and the issue is still there. It makes events practically unusable.  Unless somebody knows workaround...

**Comment by manucorporat** [comment]
When you use events, you also have to unsubscribe.

**Comment by tomgallagher** [comment]
That's interesting. You're saying that the event listener is meant to be instantiated and destroyed each time the ionview is entered or left?

**Comment by manucorporat** [comment]
@tomgallagher not exactly, I am saying each time, you run `this.events.subscribe()` you are subscribing to that event, it is not garbage collected or something.

So you would need to unsubscribe for example when the component is destroyed... `ngOnDestroy`.

**Comment by tomgallagher** [comment]
OK,  so if it's in a root page constructor, say the root page of a tab in a tabs application, when does that page get destroyed? I would have thought never.

**Comment by tomgallagher** [comment]
Just as an aside, in the Ionic 2 conference app.component.ts, the events listener is placed in the constructor. As this presumably works as expected, aren't we looking at some oddity in the relationship between events and page life cycles, as discussed here, for the use of events in other types of pages:

https://forum.ionicframework.com/t/events-and-page-lifecycles/59204
